### PR TITLE
RELATED: RAIL-3131 - Support ranking filters that reference attributes by localId

### PR DESCRIPTION
**Pull request by lupko**
-  This is not supported by tiger backend
-  Modified convertor code to tunnel attributes into the filter convertor
   so that ranking filter convertor can do lookup

JIRA: RAIL-3131

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-backend-tiger/src/convertors/toBackend/afm/MeasureConverter.ts [hunk 1]
```diff
@@ -102,7 +102,9 @@ function convertSimpleMeasureDefinition(definition: IMeasureDefinition): SimpleM
     const { measureDefinition } = definition;
 
     const filters: FilterDefinitionForSimpleMeasure[] = measureDefinition.filters
-        ? (compact(measureDefinition.filters.map(convertFilter)) as FilterDefinitionForSimpleMeasure[]) // measureDefinition.filters is IMeasureFilter, it contains only date and attribute filter, equally result contains this subset, it corresponds to type FilterDefinitionForSimpleMeasure
+        ? (compact(
+              measureDefinition.filters.map((filter) => convertFilter(filter)),
```
Comment (david-kubecka) [inline libs/sdk-backend-tiger/src/convertors/toBackend/afm/MeasureConverter.ts:106]: Why this change?

**Review by no23reason — APPROVED** [review]
LGTM

File: libs/sdk-backend-tiger/src/convertors/toBackend/afm/MeasureConverter.ts (hunk 1 above)
Comment (lupko) [reply libs/sdk-backend-tiger/src/convertors/toBackend/afm/MeasureConverter.ts:106]: TS reports it as a problem otherwise:

<html>TS2345: Argument of type '(filter0: IAbsoluteDateFilter | { relativeDateFilter: { dataSet: ObjRef; granularity: DateAttributeGranularity; from: number; to: number; }; } | ... 5 more ... | IFilterWithApplyOnResult, afmAttributes?: IAttribute[]) =&gt; NegativeAttributeFilter | ... 7 more ... | null' is not assignable to parameter of type '(value: IMeasureFilter, index: number, array: IMeasureFilter[]) =&gt; NegativeAttributeFilter | PositiveAttributeFilter | ... 6 more ... | null'.<br/>Types of parameters 'afmAttributes' and 'index' are incompatible.<br/>Type 'number' is not assignable to type 'IAttribute[] | undefined'.

Comment (lupko) [reply libs/sdk-backend-tiger/src/convertors/toBackend/afm/MeasureConverter.ts:106]: that is because map function has signature: `(value: T, index: number, array: T[])`

Comment (david-kubecka) [reply libs/sdk-backend-tiger/src/convertors/toBackend/afm/MeasureConverter.ts:106]: I see.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2924/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4562/ : SUCCESS in 14m 46s

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1156

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/862/ : SUCCESS in 1m 02s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1126/ : SUCCESS in 43s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4564/ : SUCCESS in 15m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/237/ : SUCCESS in 12m 22s (non-voting)
